### PR TITLE
[Table] Add method to return Component at a specified row, column index.

### LINKED ISSUE #2842: [Table] - `componentAt(row, column)` endpoint
There currently does not seem to be a function to get a Component at a specified location on a `Components.Table`

### PR DESCRIPTION
**Pull request by jtlan**
New API Point:
```Typescript
/**
 * Returns the Component at the specified row and column index.
 *
 * @param {number} rowIndex
 * @param {number} columnIndex
 * @returns {Component} The Component at the specified position, or null if no Component is there.
 */
componentAt(rowIndex: number, columnIndex: number): Component;
```

Fiddle: http://jsfiddle.net/pbotk0v8/

Close #2842.

### CODE REVIEW

File: src/components/table.ts [hunk 1]
```diff
@@ -81,6 +81,20 @@ export module Components {
...
+     * @param {number} rowIndex
+     * @param {number} columnIndex
+     * @returns {Component} The Component at the specified position, or null if no Component is there.
+     */
+    public componentAt(rowIndex: number, columnIndex: number) {
+      if (rowIndex < 0 || rowIndex >= this._nRows || columnIndex < 0 || columnIndex >= this._nCols) {
+        return null;
+      }
+      return this._rows[rowIndex][columnIndex];
+    };
```
Comment (bluong) [inline src/components/table.ts]: Just curious (no code action needed, necessarily): let's say that we want to add an endpoint to set the component at a specified index, would we add it through extending this endpoint?

File: src/components/table.ts [hunk 2]
```diff
@@ -81,6 +81,20 @@ export module Components {
...
 
     /**
+     * Returns the Component at the specified row and column index.
+     *
+     * @param {number} rowIndex
+     * @param {number} columnIndex
+     * @returns {Component} The Component at the specified position, or null if no Component is there.
+     */
+    public componentAt(rowIndex: number, columnIndex: number) {
+      if (rowIndex < 0 || rowIndex >= this._nRows || columnIndex < 0 || columnIndex >= this._nCols) {
```
Comment (bluong) [inline src/components/table.ts]: Use `Utils.Math.inRange` instead?

File: src/components/table.ts (hunk 1 above)
Comment (jtlan) [inline src/components/table.ts]: You mean `add()`? Or something like that? There might not be a way to name it that makes it clear this is a getter under our chosen conventions.

**Comment by crmorford** [comment]
copy a table with interactions on components http://jsfiddle.net/j5b9f60c/54/
call methods on componentAt(i. j) http://jsfiddle.net/j5b9f60c/55/

**Pull request merged**